book/6-modelling/cv-metrics.md

```diff
@@ -0,0 +1,80 @@
+
+## Mean Average Precision (mAP)
+
+There are many articles on how to calculate mAP available. It is thus instructive to instead discuss considerations when reading mAP.
+
+In discussing mAP evaluation, consider the output below:
+
+| Average Precision 	| IOU       	| Area   	| Max Detections 	| mAP   	|
+|-------------------	|-----------	|--------	|----------------	|-------	|
+| Average Precision 	| 0.50:0.95 	| all    	| 100            	| 0.326 	|
+| Average Precision 	| 0.5       	| all    	| 1000           	| 0.743 	|
+| Average Precision 	| 0.75      	| all    	| 1000           	| 0.213 	|
+| Average Precision 	| 0.50:0.95 	| small  	| 1000           	| 0.127 	|
+| Average Precision 	| 0.50:0.95 	| medium 	| 1000           	| 0.349 	|
+| Average Precision 	| 0.50:0.95 	| large  	| 1000           	| 0.406 	|
+
+Firstly, it is useful to evaluate the Intersection over Union (IOU) threshold value to understand how "tight" a bbox/mask. A higher IOU threshold value indicates that the model's inference closely overlaps with the ground truth. Generally, the mAP suffers as the IOU threshold value increases. Thus, evaluating how much these values fall would be valuable clues on how well the model may produce tight bboxes/masks.
+
+Using the outputs above, the second row shows that at IOU threshold of 0.50, the mAP is given by 0.743. Similarly, the third row showing IOU=0.75 threshold shows 0.213, a 0.53 fall in value. This indicates that the bboxes/masks produced by the model were not tight enough, and more training may be required. 
+
+Secondly, the objects being detected can be broadly separated to small/medium/large objects based on the bbox/mask size. Here, it is assumed the small/medium/large are area of the bbox/mask, and that area threshold for these sizes are clearly defined (see CV EDA article to guide the size of these annotations). 
+
+A model generally performs the worst on small objects and gets better as the size increases. Thus, evaluating this is also instructive. Considering the above outputs, the mAP is 0.127/0.349/0.406 on small/medium/large objects. Thus, it would be good to have the model be trained on more small objects in order to potentially improve its performance. 
+
+## Frames Per Second (FPS)
+
+FPS are generally measured by running multiple images through the pipeline and obtaining the amount of time taken to process each image, before converting it to frames per second. The way to do it is: 
+
+$FPS = \frac{1}{\text{Average Time Taken Per Image}}$
+
+Here, the AI Engineer would need to consider the following:
+
+FPS should be calculated based off either model's or the software architecture's inference speed. This is obvious but often overlooked. When delivering the model, the onus is on the AI Engineer to communicate how the FPS was measured to manage expectations.
```

Comment:
Suggest explaining a little about what the difference is, for new engineers who may not be aware. Something like the latter also takes into account the other parts of the inference pipeline rather than just the AI model alone?